src/components/header/_macro.spec.js

```diff
@@ -6,523 +6,348 @@ import axe from '../../tests/helpers/axe';
-    });
-
-    it('has the correct container if `wide`', () => {
-      const $ = cheerio.load(renderComponent('header', { ...EXAMPLE_HEADER_BASIC, wide: true }));
-
-      expect($('.ons-container').hasClass('ons-container--wide')).toBe(true);
-    });
-
-    it('has the correct masthead logo link', () => {
-      const $ = cheerio.load(renderComponent('header', { ...EXAMPLE_HEADER_BASIC, mastheadLogoUrl: '#0' }));
-
-      expect($('.ons-header__org-logo-link').attr('href')).toBe('#0');
-    });
-
-    it('has the default masthead logo icon', () => {
-      const faker = templateFaker();
-      const iconsSpy = faker.spy('icon');
-
-      faker.renderComponent('header', EXAMPLE_HEADER_BASIC);
-
-      expect(iconsSpy.occurrences[0].iconType).toBe('ons-logo-en');
+      test('THEN: renders with correct additional classes', () => {
+        expect($('.ons-header').hasClass('extra-class')).toBe(true);
+        expect($('.ons-header').hasClass('another-extra-class')).toBe(true);
+      });
     });
-
-    it('has the default masthead mobile logo icon', () => {
-      const faker = templateFaker();
-      const iconsSpy = faker.spy('icon');
-
-      faker.renderComponent('header', EXAMPLE_HEADER_BASIC);
-
-      expect(iconsSpy.occurrences[1].iconType).toBe('ons-logo-stacked-en');
+  });
+  describe('GIVEN: Params: fullWidth & wide', () => {
+    describe('WHEN: fullWidth parameter is set to true', () => {
+      const $ = cheerio.load(
+        renderComponent('header', {
+          ...EXAMPLE_HEADER_BASIC,
+          fullWidth: true,
+        }),
+      );
+      test('THEN: renders container with fullWidth class', () => {
+        expect($('.ons-container').hasClass('ons-container--full-width')).toBe(true);
+      });
     });
-
-    it('has the default masthead logo icon alt text', () => {
-      const faker = templateFaker();
-      const iconsSpy = faker.spy('icon');
-
-      faker.renderComponent('header', EXAMPLE_HEADER_BASIC);
-
-      expect(iconsSpy.occurrences[0].altText).toBe('Office for National Statistics homepage');
+    describe('WHEN: wide parameter is set to true', () => {
+      const $ = cheerio.load(
+        renderComponent('header', {
+          ...EXAMPLE_HEADER_BASIC,
+          wide: true,
+        }),
+      );
+      test('THEN: renders container with wide class', () => {
+        expect($('.ons-container').hasClass('ons-container--wide')).toBe(true);
+      });
     });
-
-    it('has the default masthead mobile logo icon alt text', () => {
+  });
+  describe('GIVEN: Params: mastheadLogo', () => {
+    describe('WHEN: default params are provided', () => {
       const faker = templateFaker();
       const iconsSpy = faker.spy('icon');
-
       faker.renderComponent('header', EXAMPLE_HEADER_BASIC);
 
-      expect(iconsSpy.occurrences[1].altText).toBe('Office for National Statistics logo');
+      test('THEN: renders default large logo on desktop', () => {
```

Comment:
I think we should specify "small screen" and "large screen" here rather than "mobile" and "desktop" just because there are more types of screens these days than mobile and desktop

Reply:
done!